keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/image/CurrentlyDeployedDockerImageApprover.kt

```diff
@@ -1,61 +0,0 @@
-package com.netflix.spinnaker.keel.api.titus.image
-
-import com.netflix.spinnaker.keel.api.artifacts.ArtifactType.docker
-import com.netflix.spinnaker.keel.api.titus.cluster.TitusClusterSpec
-import com.netflix.spinnaker.keel.docker.ReferenceProvider
-import com.netflix.spinnaker.keel.events.ArtifactVersionDeployed
-import com.netflix.spinnaker.keel.persistence.KeelRepository
-import com.netflix.spinnaker.keel.persistence.NoMatchingArtifactException
-import kotlinx.coroutines.runBlocking
-import org.slf4j.LoggerFactory
-import org.springframework.context.event.EventListener
-import org.springframework.stereotype.Component
-
-@Component
-class CurrentlyDeployedDockerImageApprover(
-  private val repository: KeelRepository
-) {
-  private val log = LoggerFactory.getLogger(javaClass)
-
-  @EventListener(ArtifactVersionDeployed::class)
-  fun onArtifactVersionDeployed(event: ArtifactVersionDeployed) =
-    runBlocking {
-      val resourceId = event.resourceId
-      val resource = repository.getResource(resourceId)
-      val deliveryConfig = repository.deliveryConfigFor(resourceId)
-      val env = repository.environmentFor(resourceId)
-
-      (resource.spec as? TitusClusterSpec)?.let { spec ->
```

Comment:
why can we skip the cloud providers' checks? (or maybe just a link to the corresponding PR that you've mentioned in the description)

Reply:
Oh! Before, like a month ago, there was no way to get the artifact from the resource without knowing what cloud provider it was and reaching into the spec. That's why you see the spec stuff here. But Luis put in a method to get the artifact associated with a resource. So, you no longer need to know the type of the resource. So, we don't need the cloud provider stuff in here anymore and can make this generic.

Reply:
What do you mean?

Reply:
forget about it :) nevermind